docs/mapping-network-events.asciidoc

```diff
@@ -0,0 +1,152 @@
+
+Some events may also indicate each host's role in the exchange: client or server. When this information is available, the <<ecs-client,client>> and <<ecs-server,server>> fields should be used _in addition to_ the `source` and `destination` fields. The fields and values mapped under `source`/`destination` should be copied under `client`/`server`.
+
+[float]
+==== Network event mapping example
+
+Below is a DNS network event. The source device (`192.168.86.222`) makes a DNS query, acting as the client and the DNS server is the destination (`192.168.86.1`).
+
+Note this event contains additional details that would populate additional fields (such as the <<ecs-dns>>) if this was a complete mapping example. These additional fields are omitted here to focus on the network details.
+
+[source,json]
+----
+{
+  "ts":1599775747.53056,
+  "uid":"CYqFPH3nOAa0kPxA0d",
+  "id.orig_h":"192.168.86.222",
+  "id.orig_p":54162,
+  "id.resp_h":"192.168.86.1",
+  "id.resp_p":53,
+  "proto":"udp",
+  "trans_id":28899,
+  "rtt":0.02272200584411621,
+  "query":"example.com",
+  "qclass":1,
+  "qclass_name":"C_INTERNET",
+  "qtype":1,
+  "qtype_name":"A",
+  "rcode":0,
+  "rcode_name":"NOERROR",
+  "AA":false,
+  "TC":false,
+  "RD":true,
+  "RA":true,
+  "Z":0,
+  "answers":["93.184.216.34"],
+  "TTLs":[21209.0],
+  "rejected":false
+}
+----
+
+[float]
+==== Source and destination fields
+
+First, the `source.*` and `destination.*` field sets are populated:
+
+[source,json]
+----
+  "source": {
+    "ip": "192.168.86.222",
+    "port": 54162
+  }
+----
+
+[source,json]
+----
+  "destination": {
+    "ip": "192.168.86.1",
+    "port": 53
+  }
+----
+
+[float]
+==== Client and server fields
+
+Looking back at the original event, it shows the source device is the DNS client and the destination device is the DNS server. The values mapped under `source` and `destination` are copied and mapped under `client` and `server`, respectively:
+
+[source,json]
+----
+  "client": {
+    "ip": "192.168.86.222",
+    "port": 64734
+  }
+----
+
+[source,json]
+----
+  "server": {
+    "ip": "192.168.86.1",
+    "port": 53
+  }
+----
+
+Mapping both pairs of field sets gives query visibility of the same network transaction in two ways:
+
+* `source.ip:192.168.86.222` returns all events sourced from `192.168.86.222`, regardless its role in a session
+* `client.ip:192.168.86.222` returns all events with host `192.168.86.222` acting as a client.
+
+[float]
+==== Network fields
+
+Next, the <<ecs-network,network fields>> can be populated with details about the communication between both hosts:
+
+[source,json]
+----
+  "network": {
+    "protocol": "dns",
+    "type": "ipv4",
+    "transport": "udp"
+  }
+----
+
+[float]
+==== Related fields
+
+The `related.ip` field captures all the IPs present in the event in a single array:
```

Comment:
maybe call out the specific ips to include in the related.ip field in the example?

> The `related.ip` field captures all the IPs present in the event in a single array, in this case including source.ip, destination.ip, as well as any IPs included in the dns.answer field:

Should we include e.g. pipeline code to show how to determine if a dns.answer is an IP?

Reply:
I don't think we should provide example code here. This would be fine in a blog post, but I think this will be too much of a maintenance burden.

Reply:
> maybe call out the specific ips to include in the related.ip field in the example?

Good call out. I can definitely add a little more to the explanation.

> Should we include e.g. pipeline code to show how to determine if a dns.answer is an IP?

I approached this as an implementation-neutral reference of sorts. I've kept this focused on what the final mappings should be and avoided being too prescriptive about the how.